src/reference/00-Getting-Started/07A-Task-Graph.md

```diff
@@ -0,0 +1,324 @@
+[info]   | | | | +-compile:classpathConfiguration = Task[sbt.Configuration]
+[info]   | | | | | +-compile:configuration = compile
+[info]   | | | | | +-*/*:internalConfigurationMap = <function1>
+[info]   | | | | | +-*:update = Task[sbt.UpdateReport]
+[info]   | | | | |
+....
+```
+
+When you type `compile` sbt automatically performs an `update`, for example. It
+Just Works because the values required as inputs to the `compile`
+computation require sbt to do the `update` computation first.
+
+In this way, all build dependencies in sbt are *automatic* rather than
+explicitly declared. If you use a key's value in another computation,
+then the computation depends on that key.
+
+#### Declaring dependency to other settings
+
+Next, take these two keys (from [Keys](../sxr/sbt/Keys.scala.html)):
+
+```scala
+val scalacOptions = taskKey[Seq[String]]("Options for the Scala compiler.")
+val checksums = settingKey[Seq[String]]("The list of checksums to generate and to verify for dependencies.")
+```
+
+**Note**: `scalacOptions` and `checksums` have nothing to do with each other, they
+are just two keys with the same value type, where one is a task.
+
+It is possible to compile a `build.sbt` that aliases `scalacOptions` to
+`checksums`, but not the other way. For example, this is allowed:
+
+```scala
+// The scalacOptions task may be defined in terms of the checksums setting
+scalacOptions := {
+  val cs = checksums.value
+  cs
+}
+```
+
+There is no way to go the *other* direction. That is, a setting key
+can't depend on a task key. That's because a setting key is only
+computed once on project load, so the task would not be re-run every
+time, and tasks expect to re-run every time.
+
+```scala
+// The checksums setting may not be defined in terms of the scalacOptions task
+checksums := {
+  val so = scalacOptions.value
+  so
+}
+```
+
+#### Defining a setting that depends on other settings
+
+In terms of the execution timing, we can think of the settings
+as a special tasks that evaluate during the loading time.
+
+Consider defining the project organization to be the same as the project name.
+
+```scala
+// name our organization after our project (both are SettingKey[String])
+organization := {
+  val n = name.value
+  n
+}
+```
+
+Or, set the name to the name of the project's directory:
+
+```scala
+// name is a Key[String], baseDirectory is a Key[File]
+// name the project after the directory it's inside
+name := {
+  val base = baseDirectory.value
+  base.getName
+}
+```
+
+This transforms the value of `baseDirectory` using the standard `getName`
+method of `java.io.File`.
+
+Using multiple inputs is similar. For example,
+
+```scala
+name := {
+  val old = name.value
+  val o = organization.value
+  val v = version.value
+  s"project \$old from \$o version \$v"
+}
+```
+
+This sets the name in terms of its previous value as well as the
+organization and version settings.
+
+### Inlining .value calls
```

Comment:
Yeah I'm not convinced that it's a good training wheel. I think *one* example with this artificial split - the one you give here, with all the comments - is sufficient to illustrate the point. The comments are what make it understandable at all: otherwise it's just confusing to see these trivial assignments that "appear" to do nothing